.circleci/real_config.yml

```diff
@@ -31,7 +31,7 @@ parameters:
     default: determinedai/cimg-base:latest
   machine-image:
     type: string
-    default: ubuntu-2004:202201-02
+    default: ubuntu-2004:2024.01.1
   gpu-machine-image:
     type: string
     default: ubuntu-2004-cuda-11.2:202103-01
```

Comment:
This GPU image also looks a bit older. Do you know if it ok or does it need an upgrade too? 
Edit - probably good just to upgrade now 
https://discuss.circleci.com/t/linux-cuda-deprecation-and-image-policy/48568

Reply:
I agree I [considered](https://github.com/determined-ai/determined/pull/8940/commits/4592304341e8bd424cbc8b2023bb3a83ae0a343b) it but I wasn't sure about the side effects and I want get these changes in asap

Reply:
@djanicekpach There was this discussion on Slack: https://hpe-aiatscale.slack.com/archives/C04C9JXB1C2/p1709567985738729?thread_ts=1709565408.754739&cid=C04C9JXB1C2. Aside from that specific implication, which I may look into, I'll start a branch with a bulk upgrade to a more recent Ubuntu version and see how it goes.